nautobot/extras/models/roles.py

```diff
@@ -0,0 +1,93 @@
+from django.contrib.contenttypes.models import ContentType
+from django.db import models
+
+from nautobot.core.fields import AutoSlugField
+from nautobot.core.models import BaseModel
+from nautobot.extras.fields import LimitedChoiceField
+from nautobot.extras.models import ChangeLoggedModel, CustomFieldModel, RelationshipModel
+from nautobot.extras.models.mixins import NotesMixin
+from nautobot.extras.querysets import ContentTypeRelatedQuerySet
+from nautobot.extras.utils import RoleModelsQuery, extras_features
+from nautobot.utilities.choices import ColorChoices
+from nautobot.utilities.fields import ColorField
+
+
+@extras_features(
+    "custom_fields",
+    "custom_links",
+    "custom_validators",
+    "export_templates",
+    "graphql",
+    "relationships",
+    "webhooks",
+)
+class Role(BaseModel, ChangeLoggedModel, CustomFieldModel, RelationshipModel, NotesMixin):
+    # TODO(timizuo): Create abstract model for this because Role, Status and Tags
+    #  all contain similar fields
+
+    content_types = models.ManyToManyField(
+        to=ContentType,
+        related_name="roles",
+        verbose_name="Content type(s)",
+        limit_choices_to=RoleModelsQuery(),
+        help_text="The content type(s) to which this role applies.",
+    )
+    name = models.CharField(max_length=50, unique=True)
```

Comment:
Should the uniqueness constraint be on name and content_types? I could see the use case for having multiple roles with the same name but different content types and different descriptions.

Reply:
Make sense, but the problem is i cant use `unique_together` on a `ManyToMany` field, i could possibly index `name` and add a custom validation to the `save` method. What do you think @gsnider2195